### PR TITLE
[Nereids](Variant) Implement variant type in Variant and support new sub column access method

### PR DESCRIPTION
**Pull request by eldenmoon**
## Proposed changes

The query `SELECT v["a"]["b"] from simple_var WHERE cast(v["a"]["b"] as int) = 1` 

1. During the binding stage, the expression element_at(var, "xxx") is transformed into a SlotReference with a specified path. This conversion is tracked in the StatementContext, where the parent slot is the primary key and the paths are secondary keys. This structure, known as `subColumnSlotRefMap` in the StatementContext, helps to eliminate duplicates of the same slot derived from identical paths.
2. A new rule, `BindSlotWithPaths`, is introduced in the analysis stage. This rule is responsible for converting slots with paths into their respective slot suppliers. To ensure that slots with paths are correctly associated with the appropriate `LogicalOlapScan`, an additional mapping, slotToRelation, is added to the StatementContext. This mapping links the top-level slot to its corresponding relation (i.e., `LogicalOlapScan`). Consequently, subsequent slots with paths can determine the correct `LogicalOlapScan` to merge with and modify accordingly.

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/PhysicalTVFRelation.java [hunk 1]
```diff
@@ -107,7 +107,7 @@ public String toString() {
     public List<Slot> computeOutput() {
         return function.getTable().getBaseSchema()
                 .stream()
-                .map(col -> SlotReference.fromColumn(col, ImmutableList.of()))
+                .map(col -> SlotReference.fromColumn(col, ImmutableList.of(), null))
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/PhysicalTVFRelation.java:110]: This is different from computeOutput() in LogicalTVFRelation, which pass 'this' as third param. Is this expected?

Comment (eldenmoon) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/PhysicalTVFRelation.java]: fixed

**Review by starocean999 — DISMISSED** [review]
LGTM

**Comment by eldenmoon** [comment]
run buildall

**Pull request closed without merging**